### DISCUSSION TITLE
I'm trying to convert index page.tsx into ISR page that supports query params

### DISCUSSION THREAD

**Discussion by AndonMitev** [vercel/next.js#69538] [App Router]
I'm trying to convert index page.tsx into ISR page that supports query params known in advance and i'm trying with:

```
export async function generateStaticParams() {
  const tags = await getTags();

  return tags.map((tag) => ({
    filter: tag.value,
  }));
}
```

but i don't think that works for searchParams here is my whole server component:

https://pastebin.com/0JnSABR4

Any idea how to execute this on index page.tsx that in /app  folder and supporting those search params?

**Comment by icyJoseph** [comment]
Hi, 

Yeah, if I understand your question, you are asking for this https://github.com/vercel/next.js/discussions/58181 which is not yet supported, AFAIK :/
